src/View/Widget/MultiCheckboxWidget.php

```diff
@@ -113,10 +115,26 @@ public function render(array $data, ContextInterface $context)
             'idPrefix' => null,
             'templateVars' => []
         ];
-        $out = [];
         $this->_idPrefix = $data['idPrefix'];
         $this->_clearIds();
+        return implode('', $this->_renderInputs($data, $context));
+    }
+
+    protected function _renderInputs($data, $context)
+    {
+        $out = [];
         foreach ($data['options'] as $key => $val) {
+            // Grouped inputs in a fieldset.
+            if (is_string($key) && is_array($val) && !isset($val['text'], $val['value'])) {
+                $inputs = $this->_renderInputs(['options' => $val] + $data, $context);
+                $legend = $this->_templates->format('legend', ['text' => $key]);
+                $out[] = $this->_templates->format('fieldset', [
```

Comment:
Ok. I can add new templates. New templates should be backwards compatible as custom templates are layered on top of the defaults.